sycl/source/device.cpp

```diff
@@ -16,6 +16,7 @@
 #include <sycl/info/info_desc.hpp>
 
 #include <algorithm>
+#include <typeinfo>
```

Comment:
Should we surround this include with:
```c++
#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
#include <typeinfo>
#endif
```
?

Reply:
Yes. Done. Thank.s